src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,142 @@
+import classNames from 'classnames';
 import React from 'react';
+import { getTodos } from '../../api';
+import { TodoFilter } from '../TodoFilter';
+import { TodoRandomize } from '../TodoRandomize';
 import './TodoList.scss';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
-    </div>
-  </div>
-);
+interface State {
+  todos: Todo[];
+  query: string;
+  renderStatus: Status;
+}
+
+interface Props {
+  chooseUser: (userId: number) => void;
+  selectedUserId: number;
+}
+
+enum Status {
+  all = 'all',
+  active = 'active',
+  completed = 'completed',
+}
+
+export class TodoList extends React.Component<Props, State> {
+  state: State = {
+    todos: [],
+    query: '',
+    renderStatus: Status.all,
+  };
+
+  componentDidMount() {
+    this.loadData();
+  }
+
+  componentDidUpdate(_prevProps: Props, prevState: State) {
+    if (prevState.query !== this.state.query) {
+      this.loadData();
+    }
+  }
+
+  searchByTitle = (value: string) => {
+    this.setState({ query: value });
+  };
+
+  changeShowStatus = (value: Status) => {
+    this.setState({ renderStatus: value as Status });
+  };
+
+  showTodosByStatus = () => {
+    const { renderStatus, todos } = this.state;
+
+    switch (renderStatus) {
+      case Status.all:
+        return [...todos];
+      case Status.active:
+        return [...todos].filter(todo => todo.completed === false);
+      case Status.completed:
+        return [...todos].filter(todo => todo.completed === true);
+      default:
+        return [...todos];
```

Comment:
```suggestion
      case Status.all:
      default:
        return [...todos];
```
So you will not repeat the same result